src/server/utils/wallets/getAwsKmsAccount.ts

```diff
@@ -0,0 +1,120 @@
+import { type KMSClientConfig } from "@aws-sdk/client-kms";
+import { KmsSigner } from "aws-kms-signer";
+import type { Hex, ThirdwebClient } from "thirdweb";
+import { keccak256, type Address as TwAddress } from "thirdweb";
+import { eth_sendRawTransaction, getRpcClient } from "thirdweb/rpc";
+import { serializeTransaction } from "thirdweb/transaction";
+import { toBytes } from "thirdweb/utils";
+import { type Account } from "thirdweb/wallets";
+import type {
+  SignableMessage,
+  TransactionSerializable,
+  TypedData,
+  TypedDataDefinition,
+} from "viem";
+import { hashTypedData } from "viem";
+import { getChain } from "../../../utils/chain";
+
+type SendTransactionResult = {
+  transactionHash: Hex;
+};
+
+type SendTransactionOption = TransactionSerializable & {
+  chainId: number;
+};
+
+type AwsKmsAccountOptions = {
+  keyId: string;
+  config?: KMSClientConfig;
+  client: ThirdwebClient;
+};
+
+export async function getAwsKmsAccount(
+  options: AwsKmsAccountOptions,
+): Promise<Account> {
+  const { keyId, config, client } = options;
+  const signer = new KmsSigner(keyId, config);
+
+  // Populate address immediately
+  const addressUnprefixed = await signer.getAddress();
+  const address = `0x${addressUnprefixed}` as TwAddress;
+
+  return {
+    address,
+
+    sendTransaction: async (
```

Comment:
Nit: these are meaty functions. Can we break them out into functions?

Reply:
they rely on in-scope values, so separating thme outside of the `getAwsKmsAccount` scope with extra params seems unnecessary. Can move the definition within the body of `getAwsKmsAccount` though, so the returned object is simpler to read

Reply:
I moved it inside the body, although after the change I think having it in the return object was more readable to me.
Let me know if you'd like me to revert it or something else altogether